src/progress.d

```diff
@@ -22,12 +22,14 @@ class Progress
 
 
     size_t getTerminalWidth() {
-      size_t column;
-      winsize ws;
-      if(ioctl(STDOUT_FILENO, TIOCGWINSZ, &ws) != -1) {
-        column = ws.ws_col;
+      size_t column = default_width;
+      version (CRuntime_Musl) {
```

Comment:
Is this version defined automatically on Alpine running MUSL? The other versions we need to pass to the compiler via the command line?

Reply:
@jkt628 
Can you help to provide some answers to the questions being raised?

Reply:
must be, this is all i needed to compile with LDC on Alpine.

Reply:
Since I don't have Alpine at hand, could you please answer whether the `version` is automatically defined or not? For the `version`s we use in the current code, we need to pass `-version=NoPragma -version=NoGdk -version=Notifications` to the compiler.

My question is, do we need to add `-version=CRuntime_Musl` to the compiler flags, or is this done automatically on Alpine.

Reply:
i expect it must be.  i did not define it, until your comment i did not even **know** how to define it, but the code compiles with this patch, _i.e._,
```bash
docker build -f Dockerfile-alpine .
```
succeeds.  without this patch compilation fails with:
```
src/progress.d(26): Error: undefined identifier `winsize`
src/progress.d(27): Error: undefined identifier `ioctl`
```
obviously this casts CRuntime_Musl as a required identifier, as you note.  it seems the only place it **could** come from is the compiler.

Reply:
Thanks for the confirmation